### PR TITLE
Added Huber Loss Function

### PR DESCRIPTION
**Pull request by kausthub-kannan**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: machine_learning/loss_functions/huber_loss.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    - y_pred: The predicted values.
+
+    Returns:
+    - huber_loss: The mean of Huber Loss between y_true and y_pred.
+
+    Example usage:
+    >>> true_values = np.array([0.9, 10.0, 2.0, 1.0, 5.2])
+    >>> predicted_values = np.array([0.8, 2.1, 2.9, 4.2, 5.2])
+    >>> huber_loss(true_values, predicted_values, 1.0)
+    2.1020000000000003
```
Comment (tianyizheng02) [inline machine_learning/loss_functions/huber_loss.py:34]: The true Huber loss value is 2.102. The trailing 3 is due to floating point errors. Please independently verify your tests' expected outputs, and please use the `np.isclose()` function in your tests in order to handle these floating point errors.

File: machine_learning/loss_functions/huber_loss.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    >>> true_labels = np.array([11.0, 21.0, 3.32, 4.0, 5.0])
+    >>> predicted_probs = np.array([8.3, 20.8, 2.9, 11.2, 5.0])
+    >>> huber_loss(true_labels, predicted_probs, 1.0)
+    1.8016399999999997
```
Comment (tianyizheng02) [inline machine_learning/loss_functions/huber_loss.py:38]: Likewise, this should be 1.80164

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

**Review by tianyizheng02 — COMMENTED** [review]
I also think it's worth mentioning in the comments that you're calculating the _mean_ Huber loss: you're averaging the Huber losses of each element and returning a float

File: machine_learning/loss_functions/huber_loss.py (hunk 1 above)
Comment (kausthub-kannan) [reply machine_learning/loss_functions/huber_loss.py]: I have added np.isclose() to the doctests to handle the floating point errors

**Pull request merged**